packages/react/fast-components-styles-msft/src/utilities/color/accent-fill-gradient.ts

```diff
@@ -0,0 +1,55 @@
+import { DesignSystem } from "src";
+import { ColorHSL, hslToRGB, parseColor, rgbToHSL } from "@microsoft/fast-colors";
+import { format } from "@microsoft/fast-jss-utilities";
+import {
+    FillSwatchFamily,
+    Swatch,
+    SwatchFamily,
+    SwatchFamilyResolver,
+    swatchFamilyToSwatchRecipeFactory,
+    SwatchFamilyType,
+    SwatchRecipe,
+    SwatchResolver,
+} from "./common";
+import { accentFill } from ".";
+
+function gradientSwatchRecipe<T extends SwatchFamily>(
+    type: keyof T,
+    callback: SwatchFamilyResolver<T>
+): SwatchResolver {
+    return (designSystem: DesignSystem): Swatch => {
+        const baseRecipe: SwatchRecipe = swatchFamilyToSwatchRecipeFactory<T>(
+            type,
+            callback
+        );
+        const color: Swatch = baseRecipe(designSystem);
+        const colorHsl: ColorHSL = rgbToHSL(parseColor(color));
+        let hShifted: number = colorHsl.h + 28;
+        if (hShifted > 360) {
+            hShifted -= 360;
+        }
+        const secondaryHsl: ColorHSL = ColorHSL.fromObject({
```

Comment:
So here instead of `secondaryHsl` maybe `maxHSL`. Since it's also used only once you could either avoid creating the const or do:
```
const maxColor: Swatch = hslToRGB(
    ColorHSL.fromObject({
        h: hShifted,
        s: colorHsl.s,
        l: colorHsl.l,
    })
).toStringHexRGB()
```

This way below you can do:
```
return format(
    "linear-gradient(0deg, {0} 0%, {1} 100%)",
    (a: any): string => minColor,
    (a: any): string => maxColor
)(designSystem);
```

I might have some of these return types wrong.

Reply:
I took this mostly. I went with 'shifted' instead of 'max' though. I don't think it's as much min / max as it is original / shifted (or something along those lines). For instance, if the shift were -28 instead of +28, what would be min / max at that point?

Reply:
Good point